src/openms/include/OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/IsoSpecWrapper.h

```diff
@@ -0,0 +1,357 @@
+             const std::vector<int>& atomCounts,
+             const std::vector<std::vector<double> >& isotopeMasses,
+             const std::vector<std::vector<double> >& isotopeProbabilities,
+             double p,
+             bool do_p_trim = false);
+
+    /**
+      * @brief Setup the algorithm to run on an EmpiricalFormula
+      *
+      **/
+  IsoSpecTotalProbWrapper(const EmpiricalFormula& formula, double p, bool do_p_trim = false);
+
+  virtual IsotopeDistribution run() override final;
+
+  virtual inline bool nextConf() override final { return ILG.advanceToNextConfiguration(); };
+  virtual inline Peak1D getConf() override final { return Peak1D(ILG.mass(), ILG.prob()); };
+  virtual inline double getMass() override final { return ILG.mass(); };
+  virtual inline double getIntensity() override final { return ILG.prob(); };
+  virtual inline double getLogIntensity() override final { return ILG.lprob(); };
+
+protected:
+  IsoSpec::IsoLayeredGenerator ILG;
+  };
+
+
+
+  class OPENMS_DLLAPI IsoSpecOrderedGeneratorWrapper : public IsoSpecWrapper
+  {
+  /**
+   * @brief Generate the stream of configurations, ordered from most likely to least likely.
+   *
+   * This generator walks through the entire set of isotopologues of a given molecule, allowing
+   * the user to terminate the search on the fly. The returned isotopologues are guaranteed to
+   * be generated in order of descending probability (unlike the previous two generators which
+   * make no such guarantees).
+   *
+   * This causes the algorithm to run in O(n*log(n)) and means that is it much slower than the
+   * previous two.
+   *
+   * You should only use this generator if you don't know up-front when to stop the walk through
+   * the configuration space, and need to make the decision on the fly. If you know the threshold
+   * or the total probability needed, and only need the configurations sorted, it will be much
+   * faster to generate them using one of the previous algorithms and sort them afterwards.
+   */
+public:
+    /**
+      * @brief Constructor
+      *
+      * @param isotopeNumbers A vector of how many isotopes each element has, e.g. [2, 2, 3])
+      * @param atomCounts How many atoms of each we have [e.g. 12, 6, 6 for Glucose]
+      * @param isotopeMasses Array with the individual elements isotopic masses
+      * @param isotopeProbabilities Array with the individual elements isotopic probabilities
+      *
+      **/
+  IsoSpecOrderedGeneratorWrapper(const std::vector<int>& isotopeNumbers,
+             const std::vector<int>& atomCounts,
+             const std::vector<std::vector<double> >& isotopeMasses,
+             const std::vector<std::vector<double> >& isotopeProbabilities);
+
+    /**
+      * @brief Setup the algorithm to run on an EmpiricalFormula
+      *
+      **/
+  IsoSpecOrderedGeneratorWrapper(const EmpiricalFormula& formula);
+
+  virtual IsotopeDistribution run() override final
```

Comment:
I think its fine but one could consider if it then makes sense to have a common base class